### PR TITLE
Feature: Allow multiple templates for single notification key/type

### PR DESCRIPTION
**Pull request by pranotiTechjoomla**
This MR is to support multiple templates for a single notification. 
e.g if admin wants to send different notification for course 1 and course 2 for enrollment then admin can create the template as 
`userEnroll#1`
and 
`userEnroll#2`

### CODE REVIEW

File: src/com_tjnotifications/install.tjnotification.php [hunk 1]
```diff
@@ -365,4 +376,52 @@ public function fixMenuLinks()
...
+				{
+					continue;
+				}
+
+				JFile::delete($f);
+			}
+		}
+
+		// Remove folders
+		jimport('joomla.filesystem.file');
```
Comment (thite-amol) [inline src/com_tjnotifications/install.tjnotification.php]: @pranotiTechjoomla why this is repeated?

File: src/com_tjnotifications/install.tjnotification.php [hunk 2]
```diff
@@ -365,4 +376,52 @@ public function fixMenuLinks()
...
+	 * @param   array  $removeFilesAndFolders  Array of the files and folders to be removed
+	 *
+	 * @return  void
+	 *
+	 * @since  1.1.0
+	 */
+	private function removeObsoleteFilesAndFolders($removeFilesAndFolders)
+	{
+		// Remove files
+		jimport('joomla.filesystem.file');
```
Comment (thite-amol) [inline src/com_tjnotifications/install.tjnotification.php]: @pranotiTechjoomla This is already included in the entry file

File: src/com_tjnotifications/install.tjnotification.php [hunk 3]
```diff
@@ -365,4 +376,52 @@ public function fixMenuLinks()
...
 	}
+
+	/**
+	 * Removes obsolete files and folders
+	 *
+	 * @param   array  $removeFilesAndFolders  Array of the files and folders to be removed
+	 *
+	 * @return  void
+	 *
+	 * @since  1.1.0
```
Comment (thite-amol) [inline src/com_tjnotifications/install.tjnotification.php]: @pranotiTechjoomla Use version constant

**Comment by parthlawate** [comment]
What is the use case here ?  So notification content and key is same but u want to use a different design ?

**Comment by manojLondhe** [comment]
@pranotiTechjoomla update description describing what does this PR do

**Comment by coolbung** [comment]
The client/key design already supports this and also supports cascading i.e. if a course specific template is not found, fall back to using the generic course template. 

Eg: if you use the key as `courseenroll.4` then if a specific template exists for id#4 then it will be ised otherwise it will fall back to using the generic `courseenroll` template. 

Isn't this how it works @vijaykhollam

**Comment by vijaykhollam** [comment]
Yes, It should work like as you explained only. It was not working before, MR is for same.
@pranotiTechjoomla has used '#' instead of dot(.) format.

**Pull request merged**